packages/rpc-core/src/rpc-methods/__tests__/get-balance-test.ts

```diff
@@ -0,0 +1,56 @@
+import { createJsonRpcTransport } from '@solana/rpc-transport';
+import type { SolanaJsonRpcErrorCode } from '@solana/rpc-transport/dist/types/json-rpc-transport/json-rpc-errors';
+import type { Transport } from '@solana/rpc-transport/dist/types/json-rpc-transport/json-rpc-transport-types';
+import fetchMock from 'jest-fetch-mock';
+import { createSolanaRpcApi, SolanaRpcMethods } from '../../index';
+import { Commitment } from '../common';
+import { assertIsBase58EncodedAddress } from '@solana/keys';
+
+describe('getBalance', () => {
+    let transport: Transport<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        transport = createJsonRpcTransport({
+            api: createSolanaRpcApi(),
+            url: 'http://127.0.0.1:8899',
+        });
+    });
+
+    (['confirmed', 'finalized', 'processed'] as Commitment[]).forEach(commitment => {
+        describe(`when called with \`${commitment}\` commitment`, () => {
+            it('returns a balance of zero for a new address', async () => {
+                expect.assertions(1);
+                // This key is random, don't re-use in any tests that affect balance
+                const publicKey = '4BfxgLzn6pEuVB2ynBMqckHFdYD8VNcrheDFFCB6U5TH';
+                assertIsBase58EncodedAddress(publicKey);
+                const balance = await transport.getBalance(publicKey).send();
+                expect(balance.value).toEqual(BigInt(0));
```

Comment:
I know I did this in the other tests, but it actually probably preferable to do:

```ts
await expect(transport.getBalance(publicKey).send()).resolves.toEqual(0n);
```

…or…

```ts
const balancePromise = transport.getBalance(publicKey).send();
await expect(balancePromise).resolves.toEqual(0n);
```

That way, if it rejects, we get it _in the assertion_ rather than just in the body of the test.

Reply:
Nice that makes sense

Because it's an RPC response and I just want to check the value I went with:

```ts
const balancePromise = transport.getBalance(publicKey).send();
await expect(balancePromise).resolves.toHaveProperty('value', BigInt(0));
```